### PR TITLE
Add and structure Swift SDK API reference

### PR DESCRIPTION
**Pull request by danielnugraha**
### Description

There is no API reference for the Swift SDK, user have no way of looking up what the API of the Swift SDK looks like. Also there is lack of code documentation in the Swift SDK.

### Explanation

1. Add code documentation in the Swift SDK based on the current existing documentation in Python.
2. Structuring the documentation inside the API reference as follows:

- flwr
   - Client
   - GRPC
      - Messages
   - CoreML Client
      - Data Loader
      - Layer Wrapper
      - Model Parameters

### Checklist

- [x] Implement proposed change
- [ ] Discuss the API reference structure

### CODE REVIEW

File: src/swift/flwr/Sources/Flower/Client/Client.swift [hunk 1]
```diff
@@ -7,18 +7,33 @@
...
+/// - ``getParameters()``
+/// - ``getProperties(ins:)-4u0tf``
+/// - ``fit(ins:)``
+/// - ``evaluate(ins:)``
 public protocol Client {
+    
+    /// Return the current local model parameters.
     func getParameters() -> GetParametersRes
+    
+    /// Return set of client’s properties.
```
Comment (tanertopal) [inline src/swift/flwr/Sources/Flower/Client/Client.swift:25]:
```suggestion
    /// Return set of client properties.
```

File: src/swift/flwr/Sources/Flower/CoreML/MLFlwrClientModel.swift [hunk 2]
```diff
@@ -39,6 +41,7 @@ struct MLResult {
     let accuracy: Double
 }
 
+/// A class that is responsible for loading and retrieving model parameters to and from CoreML model.
```
Comment (tanertopal) [inline src/swift/flwr/Sources/Flower/CoreML/MLFlwrClientModel.swift:44]:
```suggestion
/// A class responsible for loading and retrieving model parameters to and from the CoreML model.
```

File: src/swift/flwr/Sources/Flower/Client/Client.swift [hunk 3]
```diff
@@ -7,18 +7,33 @@
...
-/// The protocol class for the client implementation.
-/// It contains abstract functions required for processing the server statements.
-/// The expected return types are derived from the defined return structure.
+/// Protocol for Flower clients.
+///
+/// ## Topics
+///
+/// ### Functionalities
+///
+/// - ``getParameters()``
```
Comment (danieljanes) [inline src/swift/flwr/Sources/Flower/Client/Client.swift:16]: Doesn't this also take `ins`?

File: src/swift/flwr/Sources/Flower/Client/Client.swift [hunk 4]
```diff
@@ -7,18 +7,33 @@
...
+/// - ``getProperties(ins:)-4u0tf``
+/// - ``fit(ins:)``
+/// - ``evaluate(ins:)``
 public protocol Client {
+    
+    /// Return the current local model parameters.
     func getParameters() -> GetParametersRes
```
Comment (danieljanes) [inline src/swift/flwr/Sources/Flower/Client/Client.swift:23]:
```suggestion
    func getParameters(ins: GetParametersIns) -> GetParametersRes
```

File: src/swift/flwr/Sources/Flower/Common/Typing.swift [hunk 5]
```diff
@@ -88,6 +162,7 @@ public struct Status: Equatable {
     }
 }
 
+/// Parameters request for a client
```
Comment (danieljanes) [inline src/swift/flwr/Sources/Flower/Common/Typing.swift:165]: This is not a request:
```suggestion
/// Parameters message.
```

File: src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift [hunk 6]
```diff
@@ -11,11 +11,41 @@ import NIOPosix
...
 }
 
+/// Default Flower client implementation using CoreML as local training pipeline.
+///
+/// ## Topics
+///
+/// ### Usage
+///
+/// - ``init(layerWrappers:dataLoader:compiledModelUrl:)``
+/// - ``getParameters()``
```
Comment (danieljanes) [inline src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift:27]:
```suggestion
/// - ``getParameters(ins:)``
```

File: src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift [hunk 7]
```diff
@@ -51,7 +81,7 @@ public class MLFlwrClient: Client {
         }
     }
     
-    /// Parses the parameters from the local model and returns them as GetParametersRes struct
+    /// Parses the parameters from the local model and returns them as GetParametersRes struct.
     ///
     /// - Returns: Parameters from the local model
     public func getParameters() -> GetParametersRes {
```
Comment (danieljanes) [inline src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift:87]:
```suggestion
    public func getParameters(ins: GetParametersIns) -> GetParametersRes {
```

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift (hunk 6 above)
Comment (danielnugraha) [reply src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift:27]: Will be implemented in a new PR.

File: src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift (hunk 7 above)
Comment (danielnugraha) [reply src/swift/flwr/Sources/Flower/CoreML/MLFlwrClient.swift:87]: Will be implemented in a new PR.

File: src/swift/flwr/Sources/Flower/Client/Client.swift (hunk 4 above)
Comment (danielnugraha) [reply src/swift/flwr/Sources/Flower/Client/Client.swift:23]: Will be implemented in a new PR.

File: src/swift/flwr/Sources/Flower/Client/Client.swift (hunk 3 above)
Comment (danielnugraha) [reply src/swift/flwr/Sources/Flower/Client/Client.swift:16]: Will be implemented in a new PR.

**Review by tanertopal — APPROVED** [review]

**Pull request merged**